src/gui/ignorelisteditor.cpp

```diff
@@ -212,6 +215,11 @@ int IgnoreListEditor::addPattern(const QString &pattern, bool deletable, bool re
         deletableItem->setFlags(deletableItem->flags() ^ Qt::ItemIsEnabled);
     }
 
+    // most of the time we will have more than the read only rows
+    ui->removeAllPushButton->setEnabled(true);
+    if(ui->tableWidget->rowCount() == readOnlyRows)
+        ui->removeAllPushButton->setEnabled(false);
+
```

Comment:
you can remove this block for

` ui->removeAllPushButton->setEnabled(ui->tableWidget->rowCount() != readOnlyRows);`

since you are always setting it to true or false